src/snippets/article.v2.yaml

```diff
@@ -0,0 +1,118 @@
+    version:
+        type: integer
+        minimum: 1
+    type:
+        type: string
+        enum:
+          - correction
+          - editorial
+          - feature
+          - insight
+          - research-advance
+          - research-article
+          - research-communication
+          - retraction
+          - registered-report
+          - replication-study
+          - review-article
+          - scientific-correspondence
+          - short-report
+          - tools-resources
+    doi:
+        $ref: ../misc/doi.v1.yaml
+    authorLine:
+        description: |
+            For example:
+              - Ian T Baldwin
+              - Randy Schekman, Mark Patterson
+              - Lee R Berger et al.
+        type: string
+        minLength: 1
+    title:
+        type: string
+        minLength: 1
+    titlePrefix:
+        type: string
+        minLength: 1
+    stage:
+        type: string
+        enum:
+            - preview
+            - published
+    published:
+        description: |
+            Date and time this article was first published.
+            Not present if 'stage' is 'preview' and version is '1'.
+        $ref: ../misc/date-time.v1.yaml
+    versionDate:
+        description: |
+            Date and time this version of the article was published.
+            Only present if 'stage' is 'published'.
+        $ref: ../misc/date-time.v1.yaml
+    statusDate:
+        description: |
+            Date and time this article transitioned to current status.
+            Not present if 'stage' is 'preview' and version is '1'.
+        $ref: ../misc/date-time.v1.yaml
+    volume:
+        type: integer
+        minimum: 1
+    elocationId:
+        type: string
+        minLength: 1
+    pdf:
+        type: string
+        format: uri
+    subjects:
+        type: array
+        items:
+            $ref: ../snippets/subject.v1.yaml
+        uniqueItems: true
+    researchOrganisms:
+        type: array
+        items:
+            type: string
+            minLength: 1
+        uniqueItems: true
+    abstract:
+        type: object
+        properties:
+            content:
+                type: array
+                items:
+                    oneOf:
+                      - properties:
+                          type:
+                              type: string
+                              enum:
+                                - section
+                          title:
+                              title: Text
+                              type: string
+                              minLength: 1
+                          content:
+                              description: Content
+                              type: array
+                              items:
+                                  $ref: ../blocks/paragraph.v1.yaml
+                              minItems: 1
+                      - $ref: ../blocks/paragraph.v1.yaml
+                minItems: 1
+        required:
+          - content
```

Comment:
To not reuse the standard `section` block here looks like a good idea, to restrict the allowed content, at least I hope it is. The XML sample I have has some `@id` attributes on the `<sec>` tags, I wonder if adding an optional `id` attribute for the section might be something useful?

Reply:
I will add this.